### PR TITLE
chart of account ordered using shown_number + auto generate shown_num…

### PR DESCRIPTION
**Pull request by ahmadRagheb**
…ber based on tree level

### CODE REVIEW

**Comment by rmehta** [comment]
please share a screenshot / animated GIF of this feature

File: erpnext/accounts/report/financial_statements.py [hunk 1]
```diff
@@ -265,22 +265,30 @@ def add_to_list(parent, level):
...
-		if a.report_type != b.report_type and a.report_type == "Balance Sheet":
-			return -1
-		if a.root_type != b.root_type and a.root_type == "Asset":
-			return -1
-		if a.root_type == "Liability" and b.root_type == "Equity":
-			return -1
-		if a.root_type == "Income" and b.root_type == "Expense":
-			return -1
-		return 1
+		# if a.report_type != b.report_type and a.report_type == "Balance Sheet":
```
Comment (rmehta) [inline erpnext/accounts/report/financial_statements.py]: please don't keep commented code.

Also this feature should work with / without account numbers

File: erpnext/accounts/utils.py [hunk 2]
```diff
@@ -235,14 +235,15 @@ def get_count_on(account, fieldname, date):
 
 		return count
 
+#added by ahmad ragheb
```
Comment (rmehta) [inline erpnext/accounts/utils.py]: we don't add names in code, your authorship will be in the commit log

File: erpnext/accounts/utils.py [hunk 3]
```diff
@@ -251,20 +252,119 @@ def add_ac(args=None):
...
+		p = l.index("-")  # find position of the letter "-"
+		del (l[p - 1:])  # delete it
+		s = "".join(l)  # convert back to string
+
+		query = "select shown_number from `tabAccount` where parent_account = '{0}' order by shown_number DESC;".format(
+			ac.parent_account)
+		query2parent = "select shown_number from `tabAccount` where account_name = '{0}' order by shown_number DESC;".format(
+			s)
+
+		bb3 = frappe.db.sql(query, as_dict=1)
```
Comment (rmehta) [inline erpnext/accounts/utils.py]: use descriptive variable names

File: erpnext/accounts/utils.py [hunk 4]
```diff
@@ -251,20 +252,119 @@ def add_ac(args=None):
...
+		dont_have_brother_nor_father = False
+
+		# we query for parent shown_number and put it in a list
+		for key, value in enumerate(bb4):
+			r.append(value)
+
+		# if brothers and parent doesn't exist that mean account is the first node we create
+		# in a freash tree
+		if not r:
+			have_brothers = False
```
Comment (rmehta) [inline erpnext/accounts/utils.py]: `sibings` and `parent`! (lets be gender neutral)

**Review by rmehta — CHANGES_REQUESTED** [review]

**Comment by nabinhait** [comment]
Closing this as per https://github.com/frappe/erpnext/wiki/Pull-Request-Guidelines. Please reopen after making suggested changes.

**Pull request closed without merging**